ctk/common/src/main/kotlin/org/codice/compliance/verification/core/CommonDataTypeVerifier.kt

```diff
@@ -20,177 +20,80 @@ import org.codice.compliance.SAMLCore_1_3_2_a
 import org.codice.compliance.SAMLCore_1_3_3
 import org.codice.compliance.SAMLCore_1_3_4
 import org.codice.compliance.SAMLSpecRefMessage
-import org.codice.compliance.XMLDatatypesSchema_3_2_7
-import org.codice.compliance.XMLDatatypesSchema_3_2_7_1_a
-import org.codice.compliance.XMLDatatypesSchema_3_2_7_1_b
-import org.codice.compliance.XMLDatatypesSchema_3_2_7_1_c
 import org.codice.compliance.utils.TestCommon.Companion.XSI
 import org.w3c.dom.Node
 import java.net.URI
-import java.time.format.DateTimeFormatter
-import java.time.format.DateTimeParseException
 
 class CommonDataTypeVerifier {
     companion object {
-        private const val FOUR_DIGIT_YEAR_LEN = 4
-        private const val NO_HYPHEN_MESSAGE = "No hyphen was found in the date value of"
-
         var ids = mutableListOf<String>()
 
         /** 1.3 Common Data Types **/
         fun verifyCommonDataType(samlDom: Node) {
```

Comment:
✏️ Kotlin-y alternative:
```
/** 1.3 Common Data Types **/
fun verifyCommonDataType(samlDom: Node) {
    samlDom.allChildren().forEach {
        it.attributes?.getNamedItemNS(XSI, "type")?.textContent?.let { type ->
            when {
                type.contains("string") -> verifyStringValues(it)
                type.contains("anyURI") -> verifyUriValues(it)
                type.contains("dateTime") -> verifyDateTimeValues(it)
                type.contains("ID") -> verifyIdValues(it)
                else -> UnknownError("Unknown type")
            }
        }
    }
}
```

Reply:
Modified my comment. For some reason the edits I initially made didn't get saved 😭

Reply:
Updated version:
✏️ In the decryption PR, I used @brjeter's idea of having a default value for our Node.children() and Node.allChildren() extension functions. Passing in nothing will return all the children.

Using the new Node.allChildren() extension function we can simplify to:
```
/** 1.3 Common Data Types **/
fun verifyCommonDataType(samlDom: Node) {
    samlDom.allChildren().forEach {
        it.attributes?.getNamedItemNS(XSI, "type")?.textContent?.let { type ->
            when {
                type.contains("string") -> verifyStringValues(it)
                type.contains("anyURI") -> verifyUriValues(it)
                type.contains("dateTime") -> verifyDateTimeValues(it)
                type.contains("ID") -> verifyIdValues(it)
            }
        }
    }
}
```

Reply:
Changing `allChildren` to do that